modules/admin/admin-ui/src/main/resources/web/admin/apps/content-studio/js/app/wizard/ContentWizardPanel.ts

```diff
@@ -747,17 +754,28 @@ export class ContentWizardPanel extends api.app.wizard.WizardPanel<Content> {
                         this.resetWizard();
                     }
                 });
-            } else if (this.doHtmlAreasContainId(contentId.toString())) {   // Check if there are html areas in form that contain event.getContentId() that was updated
-                new GetContentByIdRequest(this.getPersistedItem().getContentId()).sendAndParse().done((content: Content) => {
-                    this.updateWizard(content, true);
-                    if (this.isEditorEnabled()) {
-                        let liveFormPanel = this.getLivePanel();
-                        liveFormPanel.skipNextReloadConfirmation(true);
-                        liveFormPanel.loadPage(false);
-                    }
-                });
+            } else {
+                let containsHandler = () => {
+                    new GetContentByIdRequest(this.getPersistedItem().getContentId()).sendAndParse().done((content: Content) => {
+                        this.updateWizard(content, true);
+                        if (this.isEditorEnabled()) {
+                            let liveFormPanel = this.getLivePanel();
+                            liveFormPanel.skipNextReloadConfirmation(true);
+                            liveFormPanel.loadPage(false);
+                        }
+                    });
+                };
+                if (this.doHtmlAreasContainId(contentId.toString()) ||
+                    this.doesFragmentContainId(this.getPersistedItem().getPage(), contentId)) {
+                    containsHandler();
+                } else {
+                    this.doImageComponentsContainId(contentId).then((contains) => {
+                        if (contains) {
+                            containsHandler();
+                        }
+                    });
+                }
```

Comment:
The logic here is getting difficult to follow. Let's add one more method, call it for example ``doPageComponentsContainId``which calls all these three methods (HtmlAreas, Fragment, Image) under the hood and returns a promise with ``contains`` (it should be defined as ``boolean`` in arguments, btw). This way we can always expand this method with logic for other components in the future. Then the call here will boil down to:
```Typescript
this.doPageComponentsContainId(contentId).then((containsId: boolean) => {
  if (containsId) {
    containsHandler();
  }
});
```